guardrail_scripts/check_pg_permissions.sql

```diff
@@ -0,0 +1,190 @@
+--- Parameters:
+--- <host>: The hostname of the PostgreSQL server.
+--- <database>: The name of the database to connect to.
+--- <username>: The username to connect with.
+--- <db_user>: The database user for which permissions are being checked.
+--- <schema_list>: A comma-separated list of schemas to check permissions on. Example 'schema1,public,schema2'.
+--- <db_instance_type>: The type of the database instance. Options are 'rds' or 'standalone'.
+--- <is_live_migration>: A flag indicating if this is a live migration (1 for true, 0 for false). If set to 0 then the script will check for permissions for an offline migration.
+--- <is_live_migration_fall_back>: A flag indicating if this is a live migration with fallback (1 for true, 0 for false). If set to 0 then the script will detect permissions for live migration with fall-forward. Should only be set to 1 when is_live_migration is also set to 1.
+--- <path_to_script>: The path to the SQL script to be executed.
+--- <path_to_output_file>: The path where the output of the script will be saved.
+
+\echo '--- Checking Variables ---'
+
+-- Check if db_user is provided
+\if :{?db_user}
+    \echo 'Database user (db_user) is provided: ':db_user
+\else
+    \echo 'Error: Database user (db_user) is not provided!'
+    \q
+\endif
+
+-- Check if schema_list is provided
+\if :{?schema_list}
+    \echo 'Schema list (schema_list) is provided: ':schema_list
+\else
+    \echo 'Error: Schema list (schema_list) is not provided!'
+    \q
+\endif
+
+-- Check if db_instance_type is provided
+\if :{?db_instance_type}
+    \echo 'Database instance type (db_instance_type) is provided: ':db_instance_type
+\else
+    \echo 'Error: Database instance type (db_instance_type) is not provided!'
+    \q
+\endif
+
+-- Check if is_live_migration is provided
+\if :{?is_live_migration}
+    \echo 'Live migration flag (is_live_migration) is provided: ':is_live_migration
+\else
+    \echo 'Error: Live migration flag (is_live_migration) is not provided!'
+    \q
+\endif
+
+-- Check if is_live_migration_fall_back is provided
+\if :{?is_live_migration_fall_back}
+    \echo 'Live migration fallback flag (is_live_migration_fall_back) is provided: ':is_live_migration_fall_back
+\else
+    \echo 'Error: Live migration fallback flag (is_live_migration_fall_back) is not provided!'
+    \q
+\endif
+
+
+--- Check USAGE permission on schemas
+\echo '--- USAGE Permission Results ---'
+SELECT
+    schema_name,
+    'USAGE' AS permission,
+    CASE WHEN has_schema_privilege(:'db_user', schema_name, 'USAGE') THEN 'Granted' ELSE 'Missing' END AS status
+FROM information_schema.schemata;
+
+-- Check SELECT permission on tables
+\echo '--- SELECT Table Permission Results ---'
+\echo 'Note that on RDS, you may get "Permission Denied" errors for pg_catalog tables (such as pg_statistic). These errors do not affect the migration and can be ignored if they show permission as "Missing".'
```

Comment:
Should we print this only if rds?

Reply:
Its just a general comment from our documentation. Maybe we can let it stay?

Reply:
If we know if it's RDS or not, I think it makes sense to selectively print this information. 
We can take this up as an enhancement.